### DISCUSSION TITLE
Context object changes value in getServerSideProps before in catch-all route

### DISCUSSION THREAD

**Discussion by LassiterJ** [vercel/next.js#12155] [Help]
I have a catch all route ([...id] in the pages folder. When getting data in getServerSideProps the context object changes when another request matches the catch all route(from the same page).In my case this is a favicon.ico request. This overwrites the data being passed as props in the associated component and errors out.  

Below is a simplified version of the getServerSideProps function I am using.

```
export async function getServerSideProps(context) {
  const getDataFromApis = async context => {
    const data = await getData(context.req.url); // same with context.params/context.query etc.

  const getData = async (path)=> {
    const client = CreateApolloClient(); // just creates a client for graphql
    try {
      const { data } = await client.query({
        query: GET_PAGE,
        variables: {
          path: path
        }
      });
      if (data) {
        return data;
      }
    } catch (err) {
      console.error( err);
    }
  };

  const data = await getDataFromApis(context);

  return { props: { data } };
}

```
Is this expected behavior? If so what may be a workaround I can try out?
If not let me know and I will create an issue.
Thanks in advance!

**Comment by justAdevTV** [comment]
I’m assuming that the data comes from a headless CMS, so shouldn’t you be using `getStaticProps` instead of `getServerSideProps`? It’s in their docs:
https://nextjs.org/docs/basic-features/data-fetching#when-should-i-use-getstaticprops

**Comment by connershoop** [comment]
I am getting the same error, one that is incredibly hard to debug, but thank @LassiterJ  for enlightening me to what was happening.

 In my getServerSideProps function, context.params is overwritten. Simply put, when the page makes a request (ex: `/static/image.png`) stemming from `<img src='/static/image.png' />` the context.params in my `getServerSideProps` are overwritten to `{param1name: 'static', param2name: 'image.png'}`  

I created an issue here: https://github.com/vercel/next.js/issues/13566
